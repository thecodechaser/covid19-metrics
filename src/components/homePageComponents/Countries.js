import React from 'react';
import PropTypes from 'prop-types';
import { FiArrowRightCircle } from '@react-icons/all-files/fi/FiArrowRightCircle';
import { NavLink } from 'react-router-dom';
import { motion } from 'framer-motion';

const Countries = (props) => {
  const { data: { Country, TotalConfirmed } } = props;
  return (
    <motion.div
      className="category-top"
      initial={{ x: '-100vw' }}
      animate={{ x: -0 }}
      transition={{ delay: 0.2, duration: 0.8 }}
      whileHover={{ scale: 1.1, zIndex: 1 }}
    >
      <NavLink
        className="nav-links"
        to={{
          pathname: `/${Country}`, Country,
        }}
      >
        <h1 className="category-country">{Country}</h1>
      </NavLink>
      <h1 className="category-infected">{TotalConfirmed}</h1>
      <NavLink to={{
        pathname: `/${Country}`, Country,
      }}
      >
        {' '}
        <FiArrowRightCircle className="arrow-icon" />
      </NavLink>
    </motion.div>

  );
};

Countries.propTypes = {
  data: PropTypes.shape({
    TotalConfirmed: PropTypes.number,
    Country: PropTypes.string,
  }).isRequired,
};

export default Countries;
