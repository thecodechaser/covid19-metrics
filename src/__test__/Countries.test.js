import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Countries from '../components/homePageComponents/Countries';

describe('Countries tests', () => {
  test('should render', () => {
    const data = {
      country: 'India',
      TotalConfirmed: 1000,
    };
    const countries = render(
      <BrowserRouter>
        <Countries data={data} />
      </BrowserRouter>,
    );

    expect(countries).toMatchSnapshot();
  });
});
