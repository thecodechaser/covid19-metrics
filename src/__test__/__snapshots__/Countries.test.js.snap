// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="category-top"
        style="transform: none;"
      >
        <a
          class="nav-links"
          href="/undefined"
        >
          <h1
            class="category-country"
          />
        </a>
        <h1
          class="category-infected"
        >
          1000
        </h1>
        <a
          class=""
          href="/undefined"
        >
           
          <svg
            class="arrow-icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <polyline
              points="12 16 16 12 12 8"
            />
            <line
              x1="8"
              x2="16"
              y1="12"
              y2="12"
            />
          </svg>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="category-top"
      style="transform: none;"
    >
      <a
        class="nav-links"
        href="/undefined"
      >
        <h1
          class="category-country"
        />
      </a>
      <h1
        class="category-infected"
      >
        1000
      </h1>
      <a
        class=""
        href="/undefined"
      >
         
        <svg
          class="arrow-icon"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <polyline
            points="12 16 16 12 12 8"
          />
          <line
            x1="8"
            x2="16"
            y1="12"
            y2="12"
          />
        </svg>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
